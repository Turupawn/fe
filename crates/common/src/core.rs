// use camino::Utf8PathBuf;

// use crate::{
//     ingot_builder::IngotBuilder,
//     input::{IngotKind, Version},
//     Core, InputDb, InputFile, InputIngot,
// };
