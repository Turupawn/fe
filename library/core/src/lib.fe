extern {
    pub fn panic() -> !
}

pub enum Option<T> {
    Some(T),
    None,
}

impl<T> Option<T> {
    pub fn none() -> Self {
        Self::None
    }

    pub fn unwrap(self) -> T {
        match self {
            Self::Some(t) => t,
            None => panic()
        }
    }
}

pub trait Default {
    fn default() -> Self
}

impl<T> Default for Option<T> {
    fn default() -> Self {
        Self::None
    }
}
