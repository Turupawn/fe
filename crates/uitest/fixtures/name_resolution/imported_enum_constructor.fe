use core::{Option, panic}

fn f() -> u32 {
    let x = Option::Some(10)
    x.unwrap() // FIXME

    let y: Option<u8> = Option::none() // FIXME
    let z: Option<u8> = Option::default()

    match x {
        Option::Some(x) => x
        Option::None => panic()
    }
}
