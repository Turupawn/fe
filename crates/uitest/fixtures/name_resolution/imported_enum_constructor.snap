---
source: crates/uitest/tests/name_resolution.rs
expression: diags
input_file: fixtures/name_resolution/imported_enum_constructor.fe
---
error[8-0029]: `unwrap` is not found
  ┌─ imported_enum_constructor.fe:5:7
  │
5 │     x.unwrap() // FIXME
  │       ^^^^^^ `unwrap` is not found in `Option`

error[8-0029]: `none` is not found
  ┌─ imported_enum_constructor.fe:7:33
  │
7 │     let y: Option<u8> = Option::none() // FIXME
  │                                 ^^^^ `none` is not found in `Option`
